Sources/CoreLocationMiddleware/CoreLocationMiddleware.swift

```diff
@@ -1,50 +1,62 @@
 import CoreLocation
 import SwiftRex
 
-public enum LocationState {
-    case unknown
-    case notAuthorized
-    case authorized(lastPosition: CLLocation?)
+public struct LocationState: Equatable {
+    var authzType: AuthzType
+    var authzStatus: CLAuthorizationStatus
+    var location: CLLocation
+    
+    public init(authzType: AuthzType = .whenInUse,
+                authzStatus: CLAuthorizationStatus,
+                location: CLLocation) {
+        self.authzType = authzType
+        self.authzStatus = authzStatus
+        self.location = location
+    }
 }
 
-public enum LocationAction {
+public enum LocationAction: Equatable {
+    // Input
     case startMonitoring
     case stopMonitoring
+    case requestAuthorizationStatus
+    case requestAuthorizationType
+    case requestPosition
+    case requestDeviceCapabilities
+    // Output
```

Comment:
So far, at least with the way I have approached it, it seems like there's a clear cut distinction between input / output. However I have a feeling that I might not be seeing the *whole* picture, specifically when it comes to the `afterReducer` closure (since I haven't used that yet).

Anyhow, I will probably check this in as-is and address issue #1 with a different PR since there will be a bit of refactoring in the Reducer and Middleware area, as you pointed out.